cmd/snap/cmd_connections.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdConnections struct {
+	clientMixin
+	All         bool `long:"all"`
+	Positionals struct {
+		Snap installedSnapName
+	} `positional-args:"true"`
+}
+
+var shortConnectionsHelp = i18n.G("List interface connections")
+var longConnectionsHelp = i18n.G(`
+The connections command lists connections between plugs and slots
+in the system.
+
+Pass --all to list unconnected plugs and slots alongside those
+already connected. Unless <snap> is provided, the listing is for all
+snaps in the system.
```

Comment:
I think we should flip this paragraph, and add a little bit to make the behaviour clearer, something like
```
Unless <snap> is provided, the listing is for connected plugs and slots
for all snaps in the system.  In this mode, pass --all to also list
unconnected plugs and slots.
```